core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -225,24 +222,55 @@ object RequestChannel extends Logging {
 
   }
 
-  /** responseAsString should only be defined if request logging is enabled */
-  class Response(val request: Request, val responseSend: Option[Send], val responseAction: ResponseAction,
-                 val responseAsString: Option[String]) {
-    request.responseCompleteTimeNanos = Time.SYSTEM.nanoseconds
-    if (request.apiLocalCompleteTimeNanos == -1L) request.apiLocalCompleteTimeNanos = Time.SYSTEM.nanoseconds
+  abstract class Response(val request: Request) {
```

Comment:
It seems more appropriate to treat this as a `ResponseAction` instead of a `Response` itself, especially given the new throttling behavior.

Reply:
I will consider taking this up in a follow-on patch.